docs/proposals/config-dump.md

```diff
@@ -0,0 +1,248 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should refer to the docs that go with that version.
+
+Documentation for other releases can be found at [releases.K8s.io](http://releases.K8s.io).
+</strong>
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+
+# Overview
+
+Here we describe a proposal for retrieving configurations (configs) that govern the behavior of any one of the Kubernetes (K8s) components. By components we mean any of the following: apiserver, scheduler, control-manager, and kubelet. We expect that these configurations be for the most part static through the life of the component but fully understand that changes from specs may affect these. Consequently, any retrieval of the configurations is to be considered true and valid at the time of retrieval and we will assume that the user is well aware of any transient behavior.   
+
+We propose several important changes:
+* new kubectl command: 
+    * `kubectl get componentconfigs <componentName> (alias compcs) --id <componentId>`
```

Comment:
right now everything is in flags. It has to be moved somewhere, whether to configmaps or some new thing. I'm claiming that we should move to configmaps and not make a new thing. Moving stuff out of flags is like 90% of the work anyway.